src/structs.rs

```diff
@@ -1,25 +1,42 @@
+use filter;
+use serde_json::Value;
+use std::slice::Iter;
+use std::iter::Enumerate;
+
 #[derive(Debug, Clone, PartialEq)]
 pub enum Criterion {
     // $
     Root,
-
+    // @
+    Element,
     // .name
     NamedChild(String),
-
+    // ?(path)
+    Filter(Vec<Criterion>),
     // .*
     AnyChild,
-
     // [123]
     IndexedChild(usize),
-
     // [10:20]
     Slice(::std::ops::Range<usize>),
-
     // [:7]
     SliceTo(::std::ops::RangeTo<usize>),
-
     // [4:]
     SliceFrom(usize),
+    // ==
+    Equal,
```

Comment:
In my opinion this comparison operators should separate entities (not Criterion)